akka-sample-fsm-java/src/main/java/sample/Hakker.java

```diff
@@ -95,12 +94,12 @@ private Hakker(ActorContext<Command> ctx, String name, ActorRef<Chopstick.Comman
    private Behavior<Command> waitForOtherChopstick(ActorRef<Chopstick.Command> chopstickToWaitFor,
                                                    ActorRef<Chopstick.Command> takenChopstick) {
       return Behaviors.receive(Command.class)
-         .onMessage(AdaptedAnswer.class, m -> m.msg.chopstick.equals(chopstickToWaitFor), msg -> {
+          .onMessage(HandleChopstickAnswer.class, m -> m.msg.isTaken() & m.msg.chopstick.equals(chopstickToWaitFor), msg -> {
```

Comment:
This is mostly just aligning with what message checks are made in the Scala sample, I think it may work without them but better safe than sorry.